open-craft/accounting
# Generate TransferWise Bulk Payment CSVs.

This PR adds the ability to generate TransferWise bulk payment CSVs based off of the data we currently have available through the accounting service.

It currently behaves the same way generating invoice PDFs does: you create a model, enter a few pieces of data (in this case just the 'sender', which is OpenCraft), hit save with some checkboxes checked, and it'll generate it for all the appropriate users who're eligible for TransferWise, and upload the CSV to Google Drive automatically too.

**Testing Instructions**:

1. Setup the necessary users, accounts, bank accounts, hourly rates, etc., just as you would for invoices.
1. Make sure a TransferWise Recipient ID is set on the (only) bank account of those users who are connected to your OpenCraft client user through an hourly rate.
1. Go to http://localhost:1786/admin/transferwise/transferwisebulkpayment/add/.
1. Mark the `sender` as the OpenCraft client user and check all the marks.
1. See that a CSV file is in your media folder, as well as in the proper folder on GDrive (same as invoices).
1. See that the CSV file has columns that match https://transferwise.com/batch#/template-instructions?useSavedRecipients=true.
1. See that the CSV file's recipient information matches what you have in the local devstack.
1. (Optional) Add more valid recipients, generate their invoices, and make the bulk payment object, and see that all rows contain proper information.
